src/jobs/jobs.controller.ts

```diff
@@ -475,18 +476,98 @@ export class JobsController {
     @Req() request: Request,
     @Body() createJobDto: CreateJobDto,
   ): Promise<JobClass> {
-    Logger.log("creating job!")
+    Logger.log("Creating job!");
     // Validate that request matches the current configuration
-    await this.validateJob(createJobDto, request);
+    await this.validateJob(createJobDto);
     // Check job authorization
-    await this.instanceAuthorization(createJobDto, request.user as JWTUser);
+    await this.instanceAuthorization(createJobDto.jobParams, request.user as JWTUser);
     // Create actual job in database
     const createdJobInstance = await this.jobsService.create(createJobDto);
     // Perform the action that is specified in the create portion of the job configuration
-    const jobServiceResponse = await this.performJobCreateAction(createdJobInstance);
+    await this.performJobCreateAction(createdJobInstance);
+    // TBD
     // Update job instance with results of job create action
-    // return await this.jobsService.update(jobServiceResponse);
-    return jobServiceResponse; // or createdJobInstance ?
+    // return await this.jobsService.statusUpdate(createdJobInstance);
+    return createdJobInstance;
+  }
+
+
+  /**
+   * Update job
+   */
+  @UseGuards(PoliciesGuard)
+  @CheckPolicies((ability: AppAbility) =>
+    ability.can(AuthOp.JobsUpdate, JobClass),
+  )
+  @Patch(":id")
+  @ApiOperation({
+    summary: "It updates an existing job.",
+    description: "It updates an existing job.",
+  })
+  @ApiBody({
+    description: "Input fields for the job to be updated",
+    required: true,
+    // schema: new UpdateJobStatusDto, // TBD
+  })
+  @ApiResponse({
+    status: HttpStatus.OK,
+    type: JobClass,
+    description: "Updated job",
+  })
+  async update(
+    @Req() request: Request,
+    @Param("id") id: string,
+    @Body() updateJobDto: UpdateJobStatusDto,
+  ): Promise<JobClass | null> {
+    Logger.log("updating job!");
+    // Find existing job
+    const currentJob = await this.jobsService.findOne({ id: id });
+    if (currentJob === null) {
+      throw new HttpException(
+        {
+          status: HttpStatus.BAD_REQUEST,
+          message: "Invalid job id.",
+        },
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+    // Check job authorization
+    await this.instanceAuthorization(currentJob.jobParams, request.user as JWTUser);
+    // Update job in database
+    const updatedJob = await this.jobsService.statusUpdate(id, updateJobDto);
+    // Emit update event
+    if (updatedJob) {
+      this.eventEmitter.emit("jobUpdated", {
+        instance: updatedJob,
+        hookState: { oldData: [updatedJob] },
+      });
+    }
+    return updatedJob;
+  }
+
+
+  /**
+   * Find job by id
+   */
+  @UseGuards(PoliciesGuard)
+  @CheckPolicies((ability: AppAbility) =>
+    ability.can(AuthOp.Read, JobClass),
+  )
+  @Get(":id")
```

Comment:
**All other endpoints** apart from `create()` and `update()`, have the following error. I log in and authorize in swagger as admin, however it seems that an admin does not have the rights to perform a GET or DELETE.
`{
  "statusCode": 403,
  "message": "Forbidden resource",
  "error": "Forbidden"
}`

Reply:
There is no DELETE on the job.
I would need to check on the GET.

Reply:
We keep create, update, get, get by id, delete. 
Get and get by id to be checked by @nitrosx, current user rights probably not defined well.
For delete, we need to restrict user access.